File: src/main/java/com/EmosewaPixel/pixellib/recipes/EnergyRecipeBuilder.java
package com.EmosewaPixel.pixellib.recipes;

public class EnergyRecipeBuilder extends AbstractRecipeBuilder<EnergyMachineRecipe, EnergyRecipeBuilder> {
    private int energyPerTick;

    public EnergyRecipeBuilder(AbstractRecipeList<EnergyMachineRecipe, EnergyRecipeBuilder> list) {
        super(list);
    }

    public EnergyRecipeBuilder energyPerTick(int energy) {
        energyPerTick = energy;
        return this;
    }

    protected int getEnergyPerTick() {
        return energyPerTick;
    }

    @Override
    public EnergyMachineRecipe build() {
        if (getInputs().size() <= getRecipeList().getMaxInputs() && getOutputs().size() <= getRecipeList().getMaxOutputs())
            return new EnergyMachineRecipe(getInputs().toArray(), getOutputs().toArray(), getTime(), energyPerTick);
        return EnergyMachineRecipe.EMPTY;
    }
}

File: src/main/java/com/emosewapixel/pixellib/recipes/EnergyRecipeBuilder.kt
package com.emosewapixel.pixellib.recipes

//This is a builder for Energy Machine Recipes
class EnergyRecipeBuilder(list: AbstractRecipeList<EnergyMachineRecipe, EnergyRecipeBuilder>) : AbstractRecipeBuilder<EnergyMachineRecipe, EnergyRecipeBuilder>(list) {
    var energyPerTick: Int = 0

    fun energyPerTick(energy: Int): EnergyRecipeBuilder {
        energyPerTick = energy
        return this
    }

    override fun build() = EnergyMachineRecipe(inputs.map { it.first }.toTypedArray(), inputs.map { it.second }.toTypedArray(), fluidInputs.toTypedArray(), outputs.map { it.first }.toTypedArray(), outputs.map { it.second }.toTypedArray(), fluidOutputs.toTypedArray(), time, energyPerTick)
}